portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -69,7 +71,26 @@ private String getComment(HttpServletRequest request) {
     }
     return value;
   }
-}
 
+  private int getCommentLimit(HttpServletRequest request) {
+    String commentLimitString = request.getParameter("limit");
+    int commentLimit;
+    if (commentLimitString == null) {
+      commentLimit = -1;
```

Comment:
A common way to handle an error gracefully is to have a default constant.

So further up in your class have something like this:
  static final int DEFUALT_COMMENT_LIMIT = 10;

Then anywhere you have a bad value you return the default.